CodeStarterCamp_Week3/Round3Step1/CoffeeShop.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  CoffeeShop.swift
+//  CodeStarterCamp_Week3
+//
+//  Created by QL Jeong on 2023/08/29.
+//
+
+import Foundation
+
+class CoffeeShop {
+    struct MenuItem: Hashable {
+        var coffee: Coffee
+        var price: Int
+    }
+
+    private(set) var name: String
+    private(set) var barista: Person?
+    private(set) var menu = [CoffeeShop.MenuItem]()
+    private(set) var pickUpTable = [Coffee]()
+    private(set) var salesAmount = 0
+
+    init(name: String, barista: Person? = nil, menu: [CoffeeShop.MenuItem] = []) {
+        self.name = name
+        self.barista = barista
+        self.menu = menu
+    }
+}
+
+// MARK: - Public
+
+extension CoffeeShop {
+    func setBarista(_ barista: Person) {
+        self.barista = barista
+    }
+
+    func setMenu(_ items: [CoffeeShop.MenuItem]) {
+        self.menu = items
+    }
+
+    func orderCoffee(_ coffee: Coffee, from client: Person) -> OrderResult {
+        if barista == nil {
+            return .isNotOpen
+        }
+
+        guard let item = getMenuItem(of: coffee) else {
+            return .isNotInMenu
+        }
+
+        if !client.expense(item.price) {
```

Comment:
여기 조건문을 읽으면서 조금 어색함이 느껴지네요!
expense 메서드의 네이밍 또는 기능 분리에 개선이 필요해 보입니다. 🤔

저는 이런 조건문을 작성할 때 잘 읽히게 하기 위해서 아래처럼 작성하곤 합니다.
(사실 앞에 ! 붙이는 것도 지양하는 편이긴 합니다.. 😅)
저의 스타일이니 참고만 해주세용~
```suggestion
        if !client.isPayable(for: item.price) {
// 또는
        if client.canPay(for: item.price) == false {
        if client.hasEnoughMoney(for: item.price) == false {
// 등등 다양한 표현이 있을 것 같네요~!
```

Reply:
이름을 변경 하였습니다.
금액의 차감 동작에 대해선 Person 이 주체가 되어야 함으로
해당 금액이 결제 가능 한지 판단 하는 내용을 일부러 분리 하지 않았습니다.